1DIP/deriv2_cgls.ipynb

```diff
@@ -59,27 +63,118 @@
     "from cil.framework import VectorData, BlockDataContainer\n",
     "from deriv2 import deriv2\n",
     "from cil.optimisation.operators import BlockOperator, GradientOperator, IdentityOperator, FiniteDifferenceOperator\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d33631d9",
+   "metadata": {},
+   "source": [
+    "### CIL version 23.0.1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "751f22b8",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "23.0.1\n"
+     ]
+    }
+   ],
+   "source": [
+    "import cil\n",
+    "print(cil.__version__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "250103e8",
+   "metadata": {},
+   "source": [
+    "We set up a 1D inverse problem, in which the forward model integrates twice.  \n",
     "\n",
+    " It is a discretization of a first kind Fredholm integral equation whose kernel K is the  Green's function for the second derivative:\n",
+    " $$\n",
+    "    K(s,t) = \\begin{cases} s(t-1)  ,  s <  t \\\\\n",
+    "              t(s-1)  ,  s \\geq t \\end{cases} $$\n",
+    "\n",
+    "and $$ \\int_0^1K(s,t)x(t)dt=b(s). $$\n",
+    "\n",
+    " For this notebook, consider the case\n",
+    "$$ b(s) = (s^3 - s)/6          , \\ \\  x(t) = t$$\n",
+    "\n",
+    " where the integral is disretised using the Galerkin method with orthonormal box functions, with interval size $1/n$.\n",
+    "\n"
```

Comment:
Mention somewhere early on that the notebook will first set up and solve basic formulation of the problem using just python/numpy, and after that using CIL.

Reply:
Done - thanks Jakob